services/cd-service/pkg/repository/transformer_test.go

```diff
@@ -1575,6 +1575,89 @@ func TestApplicationDeploymentEvent(t *testing.T) {
 				},
 			},
 		},
+		{
+			Name: "Block deployments using locks",
+			Transformers: []Transformer{
+				&CreateEnvironment{
+					Environment: "dev",
+					Config: config.EnvironmentConfig{
+						Upstream: &config.EnvironmentConfigUpstream{
+							Latest: true,
+						},
+					},
+				},
+				&CreateEnvironment{
+					Environment: "staging",
+					Config: config.EnvironmentConfig{
+						Upstream: &config.EnvironmentConfigUpstream{
+							Latest: true,
+						},
+					},
+				},
+				&CreateEnvironmentLock{
+					Environment: "staging",
+					LockId:      "lock1",
+					Message:     "lock staging",
+				},
+				&CreateEnvironmentApplicationLock{
+					Environment: "dev",
+					Application: "myapp",
+					LockId:      "lock2",
+					Message:     "lock myapp",
+				},
+				&CreateApplicationVersion{
+					Application:    "myapp",
+					SourceCommitId: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
```

Comment:
nitpick: you probably meant to have `aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa` as commit ID.

Reply:
Doesn't really matter, right?

Reply:
Yup, it doesn't, hence the "nitpick" disclaimer.